src/main/java/com/derongan/minecraft/deeperworld/config/TimeConfig.kt

```diff
@@ -0,0 +1,20 @@
+@file:UseSerializers(WorldSerializer::class)
+
+package com.derongan.minecraft.deeperworld.config
+
+import com.mineinabyss.idofront.serialization.WorldSerializer
+import com.mineinabyss.idofront.time.TimeSpan
+import com.mineinabyss.idofront.time.TimeSpanSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.UseSerializers
+import org.bukkit.World
+
+@Serializable
+class TimeConfig(
+    val updateInterval: @Serializable(with = TimeSpanSerializer::class) TimeSpan = TimeSpan(
```

Comment:
Should be able to do 1800.seconds for the TimeSpan object. Sometimes intellij forces you to do (1800).seconds for the autocomplete to show it, but you can remove the brackets after you do the import.